server/service/src/contact_form/insert/mod.rs

```diff
@@ -0,0 +1,61 @@
+use repository::{
+    contact_form_row::{ContactFormRow, ContactFormRowRepository},
+    RepositoryError, TransactionError,
+};
+mod generate;
+mod test;
+mod validate;
+use generate::{generate, GenerateInput};
+use validate::validate;
+
+use crate::service_provider::ServiceContext;
+
+#[derive(PartialEq, Debug)]
+pub enum InsertContactFormError {
+    ContactIdAlreadyExists,
+    EmailIsInvalid,
+    EmailDoesNotExist,
+    MessageDoesNotExist,
```

Comment:
Would rename these:

 - ContactFormAlreadyExists
 - `DoesNotExist` -> `NotProvided`

"does not exist" is convention for a DB record that we couldn't find, rather than an input not provided the way we expect 🙏

Reply:
Sounds good!